core/src/main/java/org/apache/calcite/rel/core/TableModify.java

```diff
@@ -71,7 +71,8 @@ public enum Operation {
     DELETE(0),
     INSERT(1),
     UPDATE(2),
-    MERGE(3);
+    MERGE(3),
+    INSERT_OVERWRITE(4);
```

Comment:
We should avoid creating a new enum. It's fine if `INSERT OVERWRITE` is part of `INSERT` with a boolean parameter to indicate if the `OVERWRITE` flag has been set.